core/src/main/scala/filodb.core/metadata/Schemas.scala

```diff
@@ -153,8 +158,13 @@ object PartitionSchema {
 
 /**
  * A Schema combines a PartitionSchema with a DataSchema, forming all the columns of a single ingestion record.
+ *
+ * The downsample member is a var because setting it at construction time can potentially result be impossible
+ * when downsample schema is same as raw schema
+ *
+ * Important Note: Serialization will be tricky since there may be loops in object graph. Avoid if possible.
```

Comment:
Have to watch for this, I sort of remembered somewhere where Schema needed to be serialized.

Reply:
ok. I'll watch out during testing.